iModelCore/ECPresentation/Source/Content/ContentSpecificationsHandler.cpp

```diff
@@ -232,10 +307,10 @@ bvector<std::unique_ptr<RelatedPropertySpecificationPaths>> ContentSpecification
     auto flatSpecs = FlattenedRelatedPropertiesSpecification::Create(params.GetRelatedPropertySpecs(), RelatedPropertiesSpecificationScopeInfo(params.GetScopeCategorySpecifications()));
     for (ContentModifierCP modifier : GetContext().GetRulesPreprocessor().GetContentModifiers())
         {
-        if (params.GetSourceClassInfo().GetSelectClass().GetClass().Is(modifier->GetSchemaName().c_str(), modifier->GetClassName().c_str()))
+        if (!modifier->GetRelatedProperties().empty())
             {
-            DiagnosticsHelpers::ReportRule(*modifier);
-            ContainerHelpers::MovePush(flatSpecs, FlattenedRelatedPropertiesSpecification::Create(modifier->GetRelatedProperties(), RelatedPropertiesSpecificationScopeInfo(modifier->GetPropertyCategories())));
+            ContainerHelpers::MovePush(flatSpecs, CreateForSelectClassFromModifiers(params.GetSourceClassInfo().GetSelectClass().GetClass(), modifier));
+            ContainerHelpers::MovePush(flatSpecs, CreateForNestedPropertiesFromModifiers(flatSpecs, modifier, GetContext().GetSchemaHelper()));

```

Comment:
Yes, otherwise we are going to have infinite loop. And if somehow we avoid infinite loop, then `priority` attribute would make no sense when used with related properties because all related properties will be still applied.

Reply:
Made a fix as discussed